src/test/java/core/basesyntax/strategy/impl/BalanceOperationTest.java

```diff
@@ -0,0 +1,62 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.model.Fruit;
+import core.basesyntax.storage.Storage;
+import core.basesyntax.strategy.OperationHandler;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+public class BalanceOperationTest {
+    private static OperationHandler balanceOperation;
+    private static Fruit fruit;
+
+    @Rule
+    public ExpectedException exceptionRule = ExpectedException.none();
+
+    @BeforeClass
+    public static void beforeClass() {
+        balanceOperation = new BalanceOperation();
+        fruit = new Fruit("apple");
+    }
+
+    @Test
+    public void execute_validData_ok() {
+        Integer expected = 100;
+        balanceOperation.execute(fruit, expected);
+        Assert.assertEquals(
+                "The method must declare the correct amount of the corresponding object.",
+                expected, Storage.stock.get(fruit));
+        expected = 70;
+        balanceOperation.execute(fruit, expected);
+        Assert.assertEquals("The method must replace the current value.",
+                expected, Storage.stock.get(fruit));
+    }
+
+    @Test
+    public void execute_nullQuantity_notOk() {
+        exceptionRule.expect(NullPointerException.class);
+        balanceOperation.execute(fruit, null);
+    }
+
+    @Test
+    public void execute_nullFruit_notOk() {
+        exceptionRule.expect(NullPointerException.class);
+        balanceOperation.execute(null, 100);
+    }
```

Comment:
```suggestion
    @Test(expected = NullPointerException.class)
    public void execute_nullFruit_notOk() {
        balanceOperation.execute(null, 100);
    }
```
better do it like this